lib/mongoid/fields/field_types.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module Mongoid
+  module Fields
+
+    # Singleton module which contains a cache for field type definitions.
+    # Custom field types can be configured.
+    module FieldTypes
+      extend self
```

Comment:
Please use `class << self` instead to define the methods only in the correct scope.

Reply:
This is a matter of preference, but will do as you say. Rubocop [has a cop for this](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/ModuleFunction) that can enforce your preferred way.